src/RequestInformation.php

```diff
@@ -147,25 +146,75 @@ public function setStreamContent(StreamInterface $value): void {
     }
 
     /**
-     * Sets the request body from a model with the specified content type.
+     * Sets the request body from a model using the specified content type.
+     *
      * @param RequestAdapter $requestAdapter The adapter service to get the serialization writer from.
      * @param string $contentType the content type.
-     * @param Parsable ...$values the models.
+     * @param Parsable $value the models.
      */
-    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable ...$values): void {
-        if (empty($values)) {
-            throw new InvalidArgumentException('$values cannot be empty.');
+    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable $value): void {
+        try {
+            $writer = $requestAdapter->getSerializationWriterFactory()->getSerializationWriter($contentType);
```

Comment:
What happens if getSerializationWriter() does not have a SerializationWriter for the contentType? I think that is a expected exception scenario. Consider adding an exception and/or message for that scenario so we can send a message that indicates we need a SerializationWriter to support the scenario. I think this comment is applicable to `setContentFromParsableCollection`, `setContentFromScalar`, and `setContentFromScalarCollection`.

Reply:
I think I see the answer to my question here: 
https://github.com/microsoft/kiota-abstractions-php/blob/2af24f9b4e94b0e4f09201c3cdfb1cabe7e53cbc/src/Serialization/SerializationWriterFactoryRegistry.php#L46
https://github.com/microsoft/kiota-abstractions-php/blob/2af24f9b4e94b0e4f09201c3cdfb1cabe7e53cbc/src/Serialization/SerializationWriterFactoryRegistry.php#L52

What action should we expect a developer to do when this is encountered in a test environment? Consider making the error message more actionable.

`'Content type ' . $contentType . ' does not have a factory to be parsed'`

could be something like

`'Content type ' . $contentType . ' does not have a factory to be parsed. Add a SerializationWriter factory to support the content type.'`

Reply:
Thanks for this. Updated the error message